b2share/modules/records/utils.py

```diff
@@ -45,3 +43,14 @@ def is_deposit(record):
         bool: True if the record is a deposit record, else False.
     """
     return record.json['$schema'].endswith('#/draft_json_schema')
+
+
+
+
+def list_db_published_records():
+    """A generator for all the published records"""
+    query = RecordMetadata.query.filter(RecordMetadata.json != None)
```

Comment:
We cab make a join with the pid table and filter on the pid type for better performances.